Build failure of aesce.c with older Clang

When compiling for aarch64, since the merge of https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/550, the build fails with Clang ≤15.x. It succeeds with GCC or with Clang ≥16.

Steps to reproduce on aarch64: just build with Clang in the default configuration.

Steps to reproduce on Ubuntu 24.04 x86_64:

1. `apt-get install gcc-13-aarch64-linux-gnu clang-15 clang-16`
2. `clang-15 -target aarch64-linux-gnu --sysroot /usr/aarch64-linux-gnu -I include -I core -I drivers/builtin/include -c drivers/builtin/src/aesce.c`

Error transcript:

```
drivers/builtin/src/aesce.c:178:5: warning: call to undeclared function 'vaeseq_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    AESCE_ENCRYPT_ROUND_X2;
    ^
drivers/builtin/src/aesce.c:164:39: note: expanded from macro 'AESCE_ENCRYPT_ROUND_X2'
#define AESCE_ENCRYPT_ROUND_X2        AESCE_ENCRYPT_ROUND; AESCE_ENCRYPT_ROUND
                                      ^
drivers/builtin/src/aesce.c:160:13: note: expanded from macro 'AESCE_ENCRYPT_ROUND'
    block = vaeseq_u8(block, vld1q_u8(keys)); \
            ^
drivers/builtin/src/aesce.c:178:5: error: assigning to 'uint8x16_t' (vector of 16 'uint8_t' values) from incompatible type 'int'
    AESCE_ENCRYPT_ROUND_X2;
    ^~~~~~~~~~~~~~~~~~~~~~
drivers/builtin/src/aesce.c:164:39: note: expanded from macro 'AESCE_ENCRYPT_ROUND_X2'
#define AESCE_ENCRYPT_ROUND_X2        AESCE_ENCRYPT_ROUND; AESCE_ENCRYPT_ROUND
                                      ^~~~~~~~~~~~~~~~~~~
drivers/builtin/src/aesce.c:160:11: note: expanded from macro 'AESCE_ENCRYPT_ROUND'
    block = vaeseq_u8(block, vld1q_u8(keys)); \
          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
…
```
(many more such errors, evidently some header isn't getting included when it should).

This is a regression since the 1.0.0 release.

Workaround: use Clang ≥16.

--- a/drivers/builtin/src/aesce.c
+++ b/drivers/builtin/src/aesce.c
@@ -16,15 +16,16 @@
 #endif
 
 #if defined(MBEDTLS_AESCE_ARCH_IS_ARMV8_A) && !defined(__ARM_FEATURE_CRYPTO)
-/* TODO: Re-consider above after https://reviews.llvm.org/D131064 merged.
- *
- * The intrinsic declaration are guarded by predefined ACLE macros in clang:
+/* The intrinsic declaration are guarded by predefined ACLE macros in clang:
  * these are normally only enabled by the -march option on the command line.
  * By defining the macros ourselves we gain access to those declarations without
  * requiring -march on the command line.
  *
  * `arm_neon.h` is included by tf_psa_crypto_common.h, so we put these defines
- * at the top of this file, before any includes.
+ * at the top of this file, before any includes. This is necessary with
+ * Clang <=15.x. With Clang 16.0 and above, these macro definitions are
+ * no longer required, but they're harmless. See
+ * https://reviews.llvm.org/D131064
  */
 #define __ARM_FEATURE_CRYPTO 1
 /* See: https://arm-software.github.io/acle/main/acle.html#cryptographic-extensions

--- a/drivers/builtin/src/aesce.c
+++ b/drivers/builtin/src/aesce.c
@@ -5,10 +5,6 @@
  *  SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
  */
 
-#include "tf_psa_crypto_common.h"
-
-#if defined(MBEDTLS_AESCE_C)
-
 #if defined(__clang__) &&  (__clang_major__ >= 4)
 
 /* Ideally, we would simply use MBEDTLS_ARCH_IS_ARMV8_A in the following #if,
@@ -41,6 +37,10 @@
 #endif
 
 #endif /* defined(__clang__) &&  (__clang_major__ >= 4) */
+
+#include "tf_psa_crypto_common.h"
+
+#if defined(MBEDTLS_AESCE_C)
 
 #include <string.h>
 

